File: H-Index/solve.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function (citations) {
    citations.sort((a, b) => a - b);
    const leng = citations.length;
    let h = 0,
        tempH;
    for (let i = 0; i < leng; i++) {
        tempH = h;
        h = Math.min(leng - i, citations[i]);
        if (h < tempH) {
            h = tempH;
            break;
        }
    }
    return h;
};

File: H-Index/solve.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    int hIndex(vector<int> &citations)
    {
        sort(citations.begin(), citations.end());
        int h = 0, tempH, leng = citations.size();
        for (int i = 0; i < leng; i++)
        {
            tempH = h;
            h = min(leng - i, citations[i]);
            h = max(h, tempH);
        }
        return h;
    }
};